### PR TITLE
New version of Cgl

### PR DESCRIPTION
**Pull request by tkralphs**

### CODE REVIEW

**Comment by tkralphs** [comment]
@dpo I made a new release of Cgl and am now trying to update that. This time, I just edited `cgl.rb` in the browser and let the CI run. Naturally, it failed, but it at least gave me the SHA, which I could then update. But now I'm running into a weird failure that seems to have nothing to do with Cgl  or the updates I made. Can you take a quick look? It could be a temporary thing, not sure, maybe I should re-run it a little later.

**Comment by tkralphs** [comment]
Ugh, I checked out the PR locally and tried to build it. The build failed in much the same way as the previous version did. When building the unit test, I got
```
make[2]: Entering directory '/tmp/cgl-20220204-3229-1eddmgs/Cgl-releases-0.60.5/Cgl/test'
g++-11 -DHAVE_CONFIG_H -I. -I`echo .` -I../src  -I`echo ./../src` -I`echo ./../src/CglFlowCover` -I`echo ./../src/CglGomory` -I`echo ./../src/CglKnapsackCover` -I`echo ./../src/CglLandP` -I`echo ./../src/CglOddHole` -I`echo ./../src/CglProbing` -I`echo ./../src/CglMixedIntegerRounding` -I`echo ./../src/CglMixedIntegerRounding2` -I`echo ./../src/CglResidualCapacity` -I`echo ./../src/CglRedSplit` -I`echo ./../src/CglRedSplit2` -I`echo ./../src/CglTwomir` -I`echo ./../src/CglClique` -I`echo ./../src/CglSimpleRounding` -I`echo ./../src/CglZeroHalf` -I/home/linuxbrew/.linuxbrew/Cellar/coinutils/2.11.6_1/include/coinutils/coin -I/home/linuxbrew/.linuxbrew/Cellar/osi/0.108.7_1/include/osi/coin -I/home/linuxbrew/.linuxbrew/Cellar/clp/1.17.6_1/include/clp/coin -I/home/linuxbrew/.linuxbrew/Cellar/coinutils/2.11.6_1/include/coinutils/coin -I/home/linuxbrew/.linuxbrew/Cellar/osi/0.108.7_1/include/osi/coin -I/home/linuxbrew/.linuxbrew/Cellar/coinutils/2.11.6_1/include/coinutils/coin  -DSAMPLEDIR=\"`echo /home/linuxbrew/.linuxbrew/opt/coin_data_sample/share/coin_data_sample/coin/Data/Sample | sed -e 's/\\\\/\\\\\\\\/g'`\" -I/home/linuxbrew/.linuxbrew/Cellar/coinutils/2.11.6_1/include/coinutils/coin -I/home/linuxbrew/.linuxbrew/Cellar/osi/0.108.7_1/include/osi/coin -I/home/linuxbrew/.linuxbrew/Cellar/clp/1.17.6_1/include/clp/coin  -I/home/linuxbrew/.linuxbrew/Cellar/coinutils/2.11.6_1/include/coinutils/coin -I/home/linuxbrew/.linuxbrew/Cellar/osi/0.108.7_1/include/osi/coin -I/home/linuxbrew/.linuxbrew/Cellar/dylp/1.10.4_1/include/dylp/coin -I/home/linuxbrew/.linuxbrew/Cellar/coinutils/2.11.6_1/include/coinutils/coin -I/home/linuxbrew/.linuxbrew/Cellar/osi/0.108.7_1/include/osi/coin  -I/home/linuxbrew/.linuxbrew/Cellar/coinutils/2.11.6_1/include/coinutils/coin -I/home/linuxbrew/.linuxbrew/Cellar/osi/0.108.7_1/include/osi/coin -I/home/linuxbrew/.linuxbrew/Cellar/vol/1.5.4_1/include/vol/coin    -O3 -pipe -DNDEBUG -Wparentheses -Wreturn-type -Wcast-qual -Wall -Wpointer-arith -Wwrite-strings -Wconversion -Wno-unknown-pragmas -Wno-long-long   -DCGL_BUILD -c -o unitTest.o unitTest.cpp
unitTest.cpp:28:10: fatal error: OsiDylpSolverInterface.hpp: No such file or directory
   28 | #include "OsiDylpSolverInterface.hpp"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [Makefile:440: unitTest.o] Error 1
```
Doing the build in debug mode and dropping into the shell right after the error, I can replicate the failure by doing `make test` in the `Cgl/test` subdirectory, but copy-pasting the exact command that fails and re-running it succeeds! This is blowing my mind.
```
ted@DESKTOP-6JTLL1U:/tmp/cgl-20220204-31690-vi3bn3/Cgl-releases-0.60.5/Cgl/test$ make test
g++-11 -DHAVE_CONFIG_H -I. -I`echo .` -I../src  -I`echo ./../src` -I`echo ./../src/CglFlowCover` -I`echo ./../src/CglGomory` -I`echo ./../src/CglKnapsackCover` -I`echo ./../src/CglLandP` -I`echo ./../src/CglOddHole` -I`echo ./../src/CglProbing` -I`echo ./../src/CglMixedIntegerRounding` -I`echo ./../src/CglMixedIntegerRounding2` -I`echo ./../src/CglResidualCapacity` -I`echo ./../src/CglRedSplit` -I`echo ./../src/CglRedSplit2` -I`echo ./../src/CglTwomir` -I`echo ./../src/CglClique` -I`echo ./../src/CglSimpleRounding` -I`echo ./../src/CglZeroHalf` -I/home/linuxbrew/.linuxbrew/Cellar/coinutils/2.11.6_1/include/coinutils/coin -I/home/linuxbrew/.linuxbrew/Cellar/osi/0.108.7_1/include/osi/coin -I/home/linuxbrew/.linuxbrew/Cellar/clp/1.17.6_1/include/clp/coin -I/home/linuxbrew/.linuxbrew/Cellar/coinutils/2.11.6_1/include/coinutils/coin -I/home/linuxbrew/.linuxbrew/Cellar/osi/0.108.7_1/include/osi/coin -I/home/linuxbrew/.linuxbrew/Cellar/coinutils/2.11.6_1/include/coinutils/coin  -DSAMPLEDIR=\"`echo /home/linuxbrew/.linuxbrew/opt/coin_data_sample/share/coin_data_sample/coin/Data/Sample | sed -e 's/\\\\/\\\\\\\\/g'`\" -I/home/linuxbrew/.linuxbrew/Cellar/coinutils/2.11.6_1/include/coinutils/coin -I/home/linuxbrew/.linuxbrew/Cellar/osi/0.108.7_1/include/osi/coin -I/home/linuxbrew/.linuxbrew/Cellar/clp/1.17.6_1/include/clp/coin  -I/home/linuxbrew/.linuxbrew/Cellar/coinutils/2.11.6_1/include/coinutils/coin -I/home/linuxbrew/.linuxbrew/Cellar/osi/0.108.7_1/include/osi/coin -I/home/linuxbrew/.linuxbrew/Cellar/dylp/1.10.4_1/include/dylp/coin -I/home/linuxbrew/.linuxbrew/Cellar/coinutils/2.11.6_1/include/coinutils/coin -I/home/linuxbrew/.linuxbrew/Cellar/osi/0.108.7_1/include/osi/coin  -I/home/linuxbrew/.linuxbrew/Cellar/coinutils/2.11.6_1/include/coinutils/coin -I/home/linuxbrew/.linuxbrew/Cellar/osi/0.108.7_1/include/osi/coin -I/home/linuxbrew/.linuxbrew/Cellar/vol/1.5.4_1/include/vol/coin    -O3 -pipe -DNDEBUG -Wparentheses -Wreturn-type -Wcast-qual -Wall -Wpointer-arith -Wwrite-strings -Wconversion -Wno-unknown-pragmas -Wno-long-long   -DCGL_BUILD -c -o unitTest.o unitTest.cpp
unitTest.cpp:28:10: fatal error: OsiDylpSolverInterface.hpp: No such file or directory
   28 | #include "OsiDylpSolverInterface.hpp"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:440: unitTest.o] Error 1
ted@DESKTOP-6JTLL1U:/tmp/cgl-20220204-31690-vi3bn3/Cgl-releases-0.60.5/Cgl/test$ g++-11 -DHAVE_CONFIG_H -I. -I`echo .` -I../src  -I`echo ./../src` -I`echo ./../src/CglFlowCover` -I`echo ./../src/CglGomory` -I`echo ./../src/CglKnapsackCover` -I`echo ./../src/CglLandP` -I`echo ./../src/CglOddHole` -I`echo ./../src/CglProbing` -I`echo ./../src/CglMixedIntegerRounding` -I`echo ./../src/CglMixedIntegerRounding2` -I`echo ./../src/CglResidualCapacity` -I`echo ./../src/CglRedSplit` -I`echo ./../src/CglRedSplit2` -I`echo ./../src/CglTwomir` -I`echo ./../src/CglClique` -I`echo ./../src/CglSimpleRounding` -I`echo ./../src/CglZeroHalf` -I/home/linuxbrew/.linuxbrew/Cellar/coinutils/2.11.6_1/include/coinutils/coin -I/home/linuxbrew/.linuxbrew/Cellar/osi/0.108.7_1/include/osi/coin -I/home/linuxbrew/.linuxbrew/Cellar/clp/1.17.6_1/include/clp/coin -I/home/linuxbrew/.linuxbrew/Cellar/coinutils/2.11.6_1/include/coinutils/coin -I/home/linuxbrew/.linuxbrew/Cellar/osi/0.108.7_1/include/osi/coin -I/home/linuxbrew/.linuxbrew/Cellar/coinutils/2.11.6_1/include/coinutils/coin  -DSAMPLEDIR=\"`echo /home/linuxbrew/.linuxbrew/opt/coin_data_sample/share/coin_data_sample/coin/Data/Sample | sed -e 's/\\\\/\\\\\\\\/g'`\" -I/home/linuxbrew/.linuxbrew/Cellar/coinutils/2.11.6_1/include/coinutils/coin -I/home/linuxbrew/.linuxbrew/Cellar/osi/0.108.7_1/include/osi/coin -I/home/linuxbrew/.linuxbrew/Cellar/clp/1.17.6_1/include/clp/coin  -I/home/linuxbrew/.linuxbrew/Cellar/coinutils/2.11.6_1/include/coinutils/coin -I/home/linuxbrew/.linuxbrew/Cellar/osi/0.108.7_1/include/osi/coin -I/home/linuxbrew/.linuxbrew/Cellar/dylp/1.10.4_1/include/dylp/coin -I/home/linuxbrew/.linuxbrew/Cellar/coinutils/2.11.6_1/include/coinutils/coin -I/home/linuxbrew/.linuxbrew/Cellar/osi/0.108.7_1/include/osi/coin  -I/home/linuxbrew/.linuxbrew/Cellar/coinutils/2.11.6_1/include/coinutils/coin -I/home/linuxbrew/.linuxbrew/Cellar/osi/0.108.7_1/include/osi/coin -I/home/linuxbrew/.linuxbrew/Cellar/vol/1.5.4_1/include/vol/coin    -O3 -pipe -DNDEBUG -Wparentheses -Wreturn-type -Wcast-qual -Wall -Wpointer-arith -Wwrite-strings -Wconversion -Wno-unknown-pragmas -Wno-long-long   -DCGL_BUILD -c -o unitTest.o unitTest.cpp
In file included from /home/linuxbrew/.linuxbrew/Cellar/vol/1.5.4_1/include/vol/coin/OsiVolSolverInterface.hpp:12,
                 from unitTest.cpp:34:
/home/linuxbrew/.linuxbrew/Cellar/vol/1.5.4_1/include/vol/coin/VolVolume.hpp:22:25: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
   22 | VolMax(register const T x, register const T y) {
      |                         ^
/home/linuxbrew/.linuxbrew/Cellar/vol/1.5.4_1/include/vol/coin/VolVolume.hpp:22:45: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
   22 | VolMax(register const T x, register const T y) {
      |                                             ^
/home/linuxbrew/.linuxbrew/Cellar/vol/1.5.4_1/include/vol/coin/VolVolume.hpp:27:25: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
   27 | VolAbs(register const T x) {
      |                         ^
/home/linuxbrew/.linuxbrew/Cellar/vol/1.5.4_1/include/vol/coin/VolVolume.hpp: In member function 'double VOL_alpha_factor::factor(const VOL_parms&, double, double)':
/home/linuxbrew/.linuxbrew/Cellar/vol/1.5.4_1/include/vol/coin/VolVolume.hpp:503:29: warning: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  503 |       register const double ll = VolAbs(lcost);
      |                             ^~
ted@DESKTOP-6JTLL1U:/tmp/cgl-20220204-31690-vi3bn3/Cgl-releases-0.60.5/Cgl/test$
```

**Comment by dpo** [comment]
It builds locally for me without disabling DyLP or Vol. I don't get the compilation error that you get. But the error in the Github Action is different; it's Homebrew-related. I'm not sure what causes it. I'll ask on the Homebrew forum.

**Comment by dpo** [comment]
Turns out it may be a bug in the brew test bot. Hopefully it will be fixed soon; you didn't do anything wrong.

> I'm building on WSL2 on Linux and I assume you're on OS X

😱  Is that a real Linux? The test bot (Ubuntu) doesn't show that build error. I'm on macOS. I've sometimes found it useful to `brew install --env=std` to debug compilation issues. Does that help in your case?

**Comment by tkralphs** [comment]
> 😱 Is that a real Linux? 
Yes, it is a real Linux running in a VM with it's own file system, a Linux kernel, etc.. Packages are installed from the same Ubuntu repos that a standard installation would install from. It's effectively like running in a Docker container, but with some more automated integration with Windows, such as the ability to copy files in and out with drag and drop, etc.

What does `brew install --env=std` do? I was actually able to run `brew test-bot` locally to fully replicate what happens on the test mchine, but I still got the failure. I can't really imagine what additional information would be helpful, I just need a theory for  how this could happen. My initial theory (which was wrong) was that it was somehow because the header is not a system header, but was included with `<...>` instead of `"..."`. I thought that was it because when I changed this and executed the failing build command, it worked! But it turns out that whenever I re-execute the failing build command, it works, even with no change! I'm close to giving up on this, since it's not really that important and disabling DyLP and Vol in the Cgl unit test is a fix until this pops up again. 

By the way, I did want to also mention that the reason DyLP and Vol are getting picked up in the first is because they are found by pkg-config, which is used by configure to get flags. As long as dependencies can be found by pkg-config, we shouldn't need specific flags to find them and in fact, those flags are just duplicating what pkg-config is doing. Does it really make sense to have  
 ```
            "--with-clp-lib=-L#{Formula["coin-or-tools/coinor/clp"].opt_lib} -lClp",
            "--with-clp-incdir=#{Formula["coin-or-tools/coinor/clp"].opt_include}/clp/coin",
```
when we don't have that for any secondary dependencies anyway and the correct flags for finding headers and libraries are already added automagically by pkg-config?

File: cgl.rb [hunk 1]
```diff
@@ -29,7 +28,10 @@ def install
             "--with-sample-datadir=#{Formula["coin-or-tools/coinor/coin_data_sample"].opt_pkgshare}/coin/Data/Sample",
             "--with-clp-lib=-L#{Formula["coin-or-tools/coinor/clp"].opt_lib} -lClp",
             "--with-clp-incdir=#{Formula["coin-or-tools/coinor/clp"].opt_include}/clp/coin",
-            "--with-dot"]
+            "--with-dot",
+            "--without-osidylp",
+            "--without-osivol"
```
Comment (dpo) [inline cgl.rb:33]:
```suggestion
            "--without-osivol"]
```

File: cgl.rb [hunk 2]
```diff
@@ -29,7 +28,10 @@ def install
...
+           ]
```
Comment (dpo) [inline cgl.rb:34]:
```suggestion
```

**Review by dpo — CHANGES_REQUESTED** [review]
needs style fixes

**Comment by dpo** [comment]
did you not label with pr-merge?

**Comment by tkralphs** [comment]
Oops, sorry, forgot about that!

**Pull request merged**